Add implementation of find command in developer guide

kangzongxian opened the pull request:
Add the parts for the find command, and our proposed implementation for a more extensible find command.

kangzongxian committed:
Update Developer Guide

Add the parts for the find command, and our proposed implementation for a more extensible find command.

kangzongxian committed:
Fix styling issues

Lan-Jingbo commented on docs/DeveloperGuide.md:
```diff
@@ -231,6 +231,47 @@ The following activity diagrams summarizes what happens when a user enters an `e
+* As the hotel manager would usually search by the Guest's name, it would be appropriate enough to search by the Guest's name. However, we do not rule out the possibility that the hotel manager might want to search by other fields as well. Hence, there could be another implementation, further elaborated on in the `Proposed` section below.
+
+**Aspect: Only matching full keywords**
+* The `find` command only matches full keywords. For example, typing in 'ali' would not match a Guest named 'Alice'. As we do not want to display possible redundant data to the hotel manager, we decided to limit the `find` command to only full keywords, so that the results displayed are more targeted.
```
Good!

Lan-Jingbo commented on docs/DeveloperGuide.md:
```diff
@@ -231,6 +231,47 @@ The following activity diagrams summarizes what happens when a user enters an `e
+### Find feature
+
+#### Implementation:
+* The `find` command takes in multiple keywords separated by spaces, and find all guests whose `Name` contain any of the keywords. The keywords are case-insensitive as well, meaning typing 'Alice' is akin to typing 'aLiCE'.
```
```suggestion
* The `find` command takes in multiple keywords separated by spaces, and find all guests whose `Name` contain any of the keywords. The keywords are case-insensitive as well, meaning typing 'Alice' is the same/similar to typing 'aLiCE'.
```
akin -> the same/similar
sometimes people may not grasp mean of slang

kangzongxian committed:
Fix grammar issues

kangzongxian committed:
Fix merge conflicts

Merge branch 'master' into update-developer-guide

* master:
  Add implementation of MarkRoomsUnclean command to Developer Guide

# Conflicts:
#	docs/DeveloperGuide.md

Lan-Jingbo reviewed (approved):
LGTM
